routers/repo/setting.go

```diff
@@ -543,7 +544,7 @@ func DeployKeysPost(ctx *context.Context, form auth.AddKeyForm) {
 		return
 	}
 
-	content, err := models.CheckPublicKeyString(form.Content)
+	content, err := models.CheckPublicKeyString(form.Content, ctx.Locale.Language())
```

Comment:
Echoing sapk's comment from above, it would be better to have `CheckPublicKeyString` return a special type of error (say `ErrSSHDisabled`) when SSH is disabled, which we can check after calling the function (like we currently do for `ErrKeyUnableVerify` on line 549).